hbase-server/src/main/java/org/apache/hadoop/hbase/wal/FSHLogProvider.java

```diff
@@ -90,13 +90,6 @@ public static Writer createWriter(final Configuration conf, final FileSystem fs,
       } else {
         LOG.debug("Error instantiating log writer.", e);
       }
-      if (writer != null) {
```

Comment:
Must we remove close logic here? I do agree @Apache9 's suggestion that we should close the writer inside initialization logic， but IMHO we should keep this close logic since the Writer is an interface. Maybe developers would implement a new XXXXWriter in the future and miss the close logic.

what about this:

in the specific writer:
```
   if (this.output != null) {
      try {
        this.output.close();
	      this.output=null
      } catch (IOException e) {
        LOG.warn("Close output failed", e);
      }
    }
```

then the real close() do nothing because `this.output` is null.

I'm not very familiar with code here, please correct me if I'm wrong.

Reply:
Thanks. I think if we change the output variable in the method, we must make synchronized/volatile for it, or else we can not avoid concurrent modify issues.

Reply:
Ok, this modification seems good to me, what about you? @Apache9